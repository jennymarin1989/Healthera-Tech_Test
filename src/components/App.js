import React, { Component } from 'react';

class App extends Component {
    render(){
        return(
            <div>
               <h1>HEALTHERA</h1> 
            </div>
        ) 
    }
}
export default App;