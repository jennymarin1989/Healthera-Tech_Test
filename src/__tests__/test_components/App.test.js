import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';


describe('App', () => {
	
	let app;

		beforeEach(()=>{
			app = shallow(<App />);
		});
	 

	it('renders correctly', () => {
		expect(app).toMatchSnapshot();
	});	

	describe("Initializes state successfully",()=> {

		it("initializes PatientPortal as false", () => {
				expect(app.state('PatientPortal')).toEqual(false);
			});

	});

	describe("Updates state  successfylly", ()=>{

		it("Changes PatientPortal state to true", ()=>{
			app.instance().handlePortalClicked();
			expect(app.state('PatientPortal')).toEqual(true);
		});

	});

	describe('Rendering components', () => {

		it("renders Header component correctly", () => {
			expect(app.find('Header').exists()).toBe(true);
		});

		it("renders SubmitButton correctly", () => {
			expect(app.find('SubmitButton').exists()).toBe(true);
		});

		it("renders Patients correctly", ()=> {
			app.setState({
				PatientPortal: true
			});
				expect(app.find('Patients').exists()).toBe(true);
		});
	});

	// describe("handleportalClicked function when click button", ()=>{
	// 	it("invokes the function when click patients button", () => {
	// 		let mockprops=jest.fn()
	// 		let app2 = shallow(<App handlePortalClicked={mockprops} />);
	// 		app2.instance().handlePortalClicked = jest.fn();
	// 		let { handlePortalClicked } = app2.instance();
	// 		expect(handlePortalClicked).toHaveBeenCalledTimes(0);
	// 		app.find('.submitButton').simulate('click'); 
	// 		expect(handleSubmit).toHaveBeenCalledTimes(1);
	// 	})
	// })

});