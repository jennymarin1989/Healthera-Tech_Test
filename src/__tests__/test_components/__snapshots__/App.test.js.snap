// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": <Row
        bsClass="row"
        componentClass="div"
>
        <Col
                bsClass="col"
                componentClass="div"
                sm={4}
                xs={6}
        >
                <Header
                        header="Healthera"
                        id="homepage"
                />
                <SubmitButton
                        handleSubmit={[Function]}
                        id="submitButton"
                        name="Patients"
                        newClass="btn btn-success"
                />
        </Col>
</Row>,
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Col
          bsClass="col"
          componentClass="div"
          sm={4}
          xs={6}
>
          <Header
                    header="Healthera"
                    id="homepage"
          />
          <SubmitButton
                    handleSubmit={[Function]}
                    id="submitButton"
                    name="Patients"
                    newClass="btn btn-success"
          />
</Col>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": Array [
            <Header
              header="Healthera"
              id="homepage"
/>,
            <SubmitButton
              handleSubmit={[Function]}
              id="submitButton"
              name="Patients"
              newClass="btn btn-success"
/>,
            null,
          ],
          "componentClass": "div",
          "sm": 4,
          "xs": 6,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "header": "Healthera",
              "id": "homepage",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleSubmit": [Function],
              "id": "submitButton",
              "name": "Patients",
              "newClass": "btn btn-success",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Row
          bsClass="row"
          componentClass="div"
>
          <Col
                    bsClass="col"
                    componentClass="div"
                    sm={4}
                    xs={6}
          >
                    <Header
                              header="Healthera"
                              id="homepage"
                    />
                    <SubmitButton
                              handleSubmit={[Function]}
                              id="submitButton"
                              name="Patients"
                              newClass="btn btn-success"
                    />
          </Col>
</Row>,
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Col
            bsClass="col"
            componentClass="div"
            sm={4}
            xs={6}
>
            <Header
                        header="Healthera"
                        id="homepage"
            />
            <SubmitButton
                        handleSubmit={[Function]}
                        id="submitButton"
                        name="Patients"
                        newClass="btn btn-success"
            />
</Col>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [
              <Header
                header="Healthera"
                id="homepage"
/>,
              <SubmitButton
                handleSubmit={[Function]}
                id="submitButton"
                name="Patients"
                newClass="btn btn-success"
/>,
              null,
            ],
            "componentClass": "div",
            "sm": 4,
            "xs": 6,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "header": "Healthera",
                "id": "homepage",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleSubmit": [Function],
                "id": "submitButton",
                "name": "Patients",
                "newClass": "btn btn-success",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
